inst/components/scss/toolbar.scss

```diff
@@ -2,35 +2,84 @@
 .bslib-toolbar {
   display: flex;
   align-items: center;
+  gap: 0;
 
   /* ---- Toolbar options ---- */
 
   &[data-align="left"] {
     margin-right: auto;
+    justify-content: start;
   }
 
   &[data-align="right"] {
     margin-left: auto;
+    justify-content: end;
   }
 
   /* ---- Adjustments to other elements ---- */
 
   // Ensures uniformity of font sizing in elements and sub-elements (e.g., input select)    
   &,
   & * {
-    font-size: 0.8rem;
+    font-size: 0.9rem;
   }
 
   &>* {
     margin-bottom: 0 !important;
     width: auto;
     align-self: center;
   }
+}
+
+/* Toolbar Select Input */
+.bslib-toolbar-input-select {
+  // Decrease this to keep header slim and leave a margin between select & border
+  padding-block: 2px; 
+
+  .form-group.shiny-input-container {
+    width: auto;
+
+    select {
+      width: auto;
+    }
+  }
+
+  /* Remove the select label visually */
+  #select-label {
+    display: none;
+  }
+
+  /* Make children take width=auto to size to content */
+  &>* {
+    width: auto;
+  }
+
+  select {
+    appearance: auto;
+    padding-block: 3px; // Decreased to slim down select to fit in header
+    /* .form-controls takes away the drop-down arrow, this ensures it shows up */
```

Comment:
I _think_ I understand what this comment is saying, but I'm not sure. Some things to clarify:

1. Which `.form-controls`? Do you mean Bootstrap's rule, Shiny's rule, or another rule in bslib?
2. I'm pretty sure `it` refers to the _drop-down arrow_, but I'm not sure what `this` refers to. I think you could probably re-order this sentence a bit to avoid two adjacent implicit references.